### DISCUSSION TITLE
How to use build_mac_app

### DISCUSSION THREAD

**Discussion by tedgonzalez** [fastlane/fastlane#18273] [Q&A]
Hi I'm trying to create a Mac catalyst for the app store using fastlane 2.175
```
    build_mac_app(
      export_options: {
        compileBitcode: false,
        method: "app-store",
        provisioningProfiles: provisioningProfilesHash,
        stripSwiftSymbols: true,
        thinning: "<none>"
      },
      installer_cert_name: "3rd Party Mac Developer Installer: COMPANY (12345ABCDE)",
      silent: false
    )
    puts "PKG_OUTPUT_PATH #{lane_context[SharedValues::PKG_OUTPUT_PATH]}"
```
It creates an IPA file but not pkg file.
PKG_OUTPUT_PATH is nil

Any ideas?

**Comment by tedgonzalez** [comment]
This is fixed by specifying 

`destination: "generic/platform=macOS,variant=Mac Catalyst",`

**Comment by Pacocha24** [comment]
I currently have an iPhone 5s, and so far that's all I have in terms of Apple Devices.

I'm not too familiar with how XCode really works, but I'm 99% sure I need a Mac, and I really can't afford it right now. I'm soon to graduate with a Programming degree, and I may want to work for Apple and developing their apps/iOS and more.

What are my best options when it comes to Programming for Apple? Should I just cut my loses and give up, at least for now?



[MyCCPay](https://www.myccpay.pro/)

**Reply by tedgonzalez** [reply]
Theoretically, you can use swift on linux and learn swift. Then when you need xcode, you can rent a mac in cloud like https://www.macstadium.com/ and https://aws.amazon.com/ec2/instance-types/mac/ but those are not cheap.
If you apply to AWS Activate Founders https://aws.amazon.com/activate/founders/, you can get 1000$ but that's only 34 days of mac rental.
Then maybe when you run out of credits, build your app via github actions or circle ci then deploy it to browserstack to install on their simulators or testflight to install on your device.

Buying a mac is the most convenient
